src/plugins/data_source_management/public/components/create_data_source_wizard/create_data_source_wizard.tsx

```diff
@@ -43,65 +38,20 @@ const CreateDataSourceWizard: React.FunctionComponent<CreateDataSourceWizardProp
   });
 
   /* Handle submit - create data source*/
-  const handleSubmit = async ({
-    title,
-    description,
-    endpoint,
-    credentialId,
-    credentialType,
-    newCredential,
-  }: CreateDataSourceFormType) => {
+  const handleSubmit = async (attributes: DataSourceAttributes) => {
     setIsLoading(true);
     try {
-      /* Create new credential, if user selects that option*/
-      if (credentialType === CredentialSourceType.CreateCredential && newCredential?.title) {
-        credentialId = await createCredential(newCredential);
-      }
-
-      const references = [];
-      const attributes = {
-        title,
-        description,
-        endpoint,
-        noAuth: credentialType === CredentialSourceType.NoAuth,
-      };
-
-      if (credentialId) {
-        references.push({ id: credentialId, type: 'credential', name: 'credential' });
-      }
-      const options = { references };
-
-      await createSingleDataSource(savedObjects.client, attributes, options);
-
+      await createSingleDataSource(savedObjects.client, attributes);
       props.history.push('');
     } catch (e) {
+      setIsLoading(false);
```

Comment:
Does this mean we only remove the loading when error but not happy case?

Reply:
Yes .. as we are navigating away from the page ... Loading false is not required ... 

Also react will error out if we try to setState and immediately navigate to another page ..